### PR TITLE
added optional prefix and whitelist for pseudo classes to use

### PR DESCRIPTION
**Pull request by grouchal**
I have added to configuration parameters to the project so that alternate prefixes can be used. This is so that the plugin can be used with our kss based style guide - but could be useful for others too.

I have also added a whitelist for pseudo classes so that only the wanted classes are converted.

I have added tests for both new options.

### CODE REVIEW

**Comment by giuseppeg** [comment]
@grouchal hey there, thank you for submitting this PR!

I like the idea of making `prefix` configurable. Usually I prefer to have very scoped branches that fix a single thing at the time. Can you put a new PR together to only add the prefix?

Regarding the whitelist, there is already a `blacklist` option so maybe you can use some reverse logic there and blacklist the ones that you don't want to compile? I would like to avoid to add more logic.

Regarding the variables refactoring I don't think that it is strictly necessary because `pseudo` is a local variable within the `map` callback/scope.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 npm-debug.log
+.idea
```
Comment (giuseppeg) [inline .gitignore:3]: This shouldn't be here.
You can create a global gitignore https://help.github.com/articles/ignoring-files/#create-a-global-gitignore
or add it to your local repo ignores: `.git/info/exclude`

**Comment by giuseppeg** [comment]
fyi I just pushed a fix so you should pull master

**Comment by grouchal** [comment]
Hi - thanks for all of the feedback, will try and address your points below:
1. Whitelist: We have tried to use the blacklist - unfortunately it doesn't work as expected (perhaps it is a bug) we can not use the blacklist to provide a list of pseudo elements that should not be converted - for example :not . When this was done it was clear that the blacklist works by comparing the entire selector and not just the pseudo elements. If the blacklist is supposed to be the opposite of the whitelist that I have implemented there is a definite bug.  In our use case we are only really interested in pseudo classes: hover, active, focus and it is easier to use a whitelist than a blacklist.
2. Please deal with the refactoring as you want.
3. Please take the .idea out of the .gitignore and I will follow the advice in your link.

Looking forward to seeing the changes in the published module and hopefully a solution to the whitelist / blacklist issue

**Comment by giuseppeg** [comment]
<blockquote>we can not use the blacklist to provide a list of pseudo elements that should not be converted</blockquote>

Oh I see what you mean, you need to provide the full selector in order to blacklist it eg. `.foo:not(.bar)`.
We could add one more condition to [line 47](https://github.com/giuseppeg/postcss-pseudo-classes/blob/master/index.js#L47):

``` javascript
// { ':not': true }
if (blacklist[pseudo] || pseudo.match(/^::/)) {
  return pseudo;
}
```

By the way is there any specific reason why you want the whitelist? Is the plugin slow?

**Comment by grouchal** [comment]
The reason we want a whitelist is so that we don't have any unforeseen side-effects from strange pseudo class combinations in different browsers. The approach seems simpler to me to convert only the ones we want rather than having to keep adding ones we don't want to the black list. I think both approaches have different use case. We are using the version with a whitelist in one of our projects and it is working very well.

**Comment by giuseppeg** [comment]
<blockquote>The approach seems simpler to me to convert only the ones we want rather than having to keep adding ones we don't want to the black list.</blockquote>

Fair enough, if you can open an issue I will implement it.
Maybe we could just call the option `restrictTo`?

Finally I think that your version doesn't work with pseudo classes that have parenthesis like `:nth-child(...)` unless you are explicit i.e. `:nth-child(2n-1)`. How should it work in this case?
Either ways we need a test for that case.

**Comment by giuseppeg** [comment]
@grouchal for the record I've implemented the "whitelist" in #3

**Comment by grouchal** [comment]
@giuseppeg hi - great to hear - not got time at the moment to support you - but will be great to hear when both of the requirements are somehow implemented in the master version.

**Comment by giuseppeg** [comment]
fixed in #3

**Pull request closed without merging**